--- ./ijkplayer-android//android/contrib/tools/do-compile-ffmpeg.sh.orig.bak	2019-04-10 17:14:52.000000000 +0800
+++ ./ijkplayer-android//android/contrib/tools/do-compile-ffmpeg.sh	2019-04-10 17:14:52.000000000 +0800
@@ -52,13 +52,17 @@
 FF_DEP_LIBSOXR_INC=
 FF_DEP_LIBSOXR_LIB=
 
+FF_DEP_X264_INC=
+FF_DEP_X264_LIB=
+
 FF_CFG_FLAGS=
 
 FF_EXTRA_CFLAGS=
 FF_EXTRA_LDFLAGS=
 FF_DEP_LIBS=
 
-FF_MODULE_DIRS="compat libavcodec libavfilter libavformat libavutil libswresample libswscale"
+#FF_MODULE_DIRS="compat libavcodec libavfilter libavformat libavutil libswresample libswscale"
+FF_MODULE_DIRS="compat libavcodec libavresample libavfilter libavformat libavutil libswresample libswscale"
 FF_ASSEMBLER_SUB_DIRS=
 
 
@@ -79,6 +83,7 @@
     FF_BUILD_NAME=ffmpeg-armv7a
     FF_BUILD_NAME_OPENSSL=openssl-armv7a
     FF_BUILD_NAME_LIBSOXR=libsoxr-armv7a
+    FF_BUILD_NAME_LIBX264=x264-armv7a
     FF_SOURCE=$FF_BUILD_ROOT/$FF_BUILD_NAME
 
     FF_CROSS_PREFIX=arm-linux-androideabi
@@ -97,6 +102,7 @@
     FF_BUILD_NAME=ffmpeg-armv5
     FF_BUILD_NAME_OPENSSL=openssl-armv5
     FF_BUILD_NAME_LIBSOXR=libsoxr-armv5
+    FF_BUILD_NAME_LIBX264=x264-armv5
     FF_SOURCE=$FF_BUILD_ROOT/$FF_BUILD_NAME
 
     FF_CROSS_PREFIX=arm-linux-androideabi
@@ -113,6 +119,7 @@
     FF_BUILD_NAME=ffmpeg-x86
     FF_BUILD_NAME_OPENSSL=openssl-x86
     FF_BUILD_NAME_LIBSOXR=libsoxr-x86
+    FF_BUILD_NAME_LIBX264=x264-x86
     FF_SOURCE=$FF_BUILD_ROOT/$FF_BUILD_NAME
 
     FF_CROSS_PREFIX=i686-linux-android
@@ -131,6 +138,7 @@
     FF_BUILD_NAME=ffmpeg-x86_64
     FF_BUILD_NAME_OPENSSL=openssl-x86_64
     FF_BUILD_NAME_LIBSOXR=libsoxr-x86_64
+    FF_BUILD_NAME_LIBX264=x264-x86_64
     FF_SOURCE=$FF_BUILD_ROOT/$FF_BUILD_NAME
 
     FF_CROSS_PREFIX=x86_64-linux-android
@@ -149,6 +157,7 @@
     FF_BUILD_NAME=ffmpeg-arm64
     FF_BUILD_NAME_OPENSSL=openssl-arm64
     FF_BUILD_NAME_LIBSOXR=libsoxr-arm64
+    FF_BUILD_NAME_LIBX264=x264-arm64
     FF_SOURCE=$FF_BUILD_ROOT/$FF_BUILD_NAME
 
     FF_CROSS_PREFIX=aarch64-linux-android
@@ -184,6 +193,8 @@
 FF_DEP_OPENSSL_LIB=$FF_BUILD_ROOT/build/$FF_BUILD_NAME_OPENSSL/output/lib
 FF_DEP_LIBSOXR_INC=$FF_BUILD_ROOT/build/$FF_BUILD_NAME_LIBSOXR/output/include
 FF_DEP_LIBSOXR_LIB=$FF_BUILD_ROOT/build/$FF_BUILD_NAME_LIBSOXR/output/lib
+FF_DEP_X264_INC=$FF_BUILD_ROOT/build/$FF_BUILD_NAME_LIBX264/output/include
+FF_DEP_X264_LIB=$FF_BUILD_ROOT/build/$FF_BUILD_NAME_LIBX264/output/lib
 
 case "$UNAME_S" in
     CYGWIN_NT-*)
@@ -258,7 +269,17 @@
     FF_DEP_LIBS="$FF_DEP_LIBS -L${FF_DEP_LIBSOXR_LIB} -lsoxr"
 fi
 
-FF_CFG_FLAGS="$FF_CFG_FLAGS $COMMON_FF_CFG_FLAGS"
+# with x264
+if [ -f "${FF_DEP_X264_LIB}/libx264.a" ]; then
+    echo "x264 detected"
+ 		FF_CFG_FLAGS="$FF_CFG_FLAGS --enable-nonfree --enable-gpl"
+    FF_CFG_FLAGS="$FF_CFG_FLAGS --enable-libx264 --enable-encoder=libx264"
+
+    FF_CFLAGS="$FF_CFLAGS -I${FF_DEP_X264_INC}"
+    FF_DEP_LIBS="$FF_DEP_LIBS -L${FF_DEP_X264_LIB} -lx264"
+fi
+
+FF_CFG_FLAGS="$COMMON_FF_CFG_FLAGS $FF_CFG_FLAGS"
 
 #--------------------
 # Standard options:
@@ -301,6 +322,7 @@
 if [ -f "./config.h" ]; then
     echo 'reuse configure'
 else
+		echo "./configure $FF_CFG_FLAGS"
     which $CC
     ./configure $FF_CFG_FLAGS \
         --extra-cflags="$FF_CFLAGS $FF_EXTRA_CFLAGS" \
@@ -326,7 +348,7 @@
 echo "--------------------"
 echo $FF_EXTRA_LDFLAGS
 
-FF_C_OBJ_FILES=
+FF_C_OBJ_FILES="fftools/cmdutils.o fftools/ffmpeg_filter.o fftools/ffmpeg_opt.o fftools/ffmpeg.o fftools/ffmpeg_hw.o"
 FF_ASM_OBJ_FILES=
 for MODULE_DIR in $FF_MODULE_DIRS
 do
@@ -346,6 +368,7 @@
     done
 done
 
+
 $CC -lm -lz -shared --sysroot=$FF_SYSROOT -Wl,--no-undefined -Wl,-z,noexecstack $FF_EXTRA_LDFLAGS \
     -Wl,-soname,libijkffmpeg.so \
     $FF_C_OBJ_FILES \
