*** libx264.c	2019-04-09 17:27:18.000000000 +0800
--- /Users/Bingchen/project_android/gsmp/ijkplayer-android/android/contrib/ffmpeg-armv5/libavcodec/libx264.c	2019-04-10 15:49:12.000000000 +0800
***************
*** 279,285 ****
--- 279,289 ----
  
      x264_picture_init( &x4->pic );
      x4->pic.img.i_csp   = x4->params.i_csp;
+ #if X264_BUILD >= 153
+     if (x4->params.i_bitdepth > 8)
+ #else
      if (x264_bit_depth > 8)
+ #endif
          x4->pic.img.i_csp |= X264_CSP_HIGH_DEPTH;
      x4->pic.img.i_plane = avfmt2_num_planes(ctx->pix_fmt);
  
***************
*** 490,495 ****
--- 494,503 ----
      x4->params.p_log_private        = avctx;
      x4->params.i_log_level          = X264_LOG_DEBUG;
      x4->params.i_csp                = convert_pix_fmt(avctx->pix_fmt);
+     
+ #if X264_BUILD >= 153
+     x4->params.i_bitdepth           = av_pix_fmt_desc_get(avctx->pix_fmt)->comp[0].depth;
+ #endif
  
      PARSE_X264_OPT("weightp", wpredp);
  
***************
*** 878,883 ****
--- 886,911 ----
      AV_PIX_FMT_NV20,
      AV_PIX_FMT_NONE
  };
+ 
+ static const enum AVPixelFormat pix_fmts_all[] = {
+     AV_PIX_FMT_YUV420P,
+     AV_PIX_FMT_YUVJ420P,
+     AV_PIX_FMT_YUV422P,
+     AV_PIX_FMT_YUVJ422P,
+     AV_PIX_FMT_YUV444P,
+     AV_PIX_FMT_YUVJ444P,
+     AV_PIX_FMT_NV12,
+     AV_PIX_FMT_NV16,
+ #ifdef X264_CSP_NV21
+     AV_PIX_FMT_NV21,
+ #endif
+     AV_PIX_FMT_YUV420P10,
+     AV_PIX_FMT_YUV422P10,
+     AV_PIX_FMT_YUV444P10,
+     AV_PIX_FMT_NV20,
+     AV_PIX_FMT_NONE
+ };
+ 
  #if CONFIG_LIBX264RGB_ENCODER
  static const enum AVPixelFormat pix_fmts_8bit_rgb[] = {
      AV_PIX_FMT_BGR0,
***************
*** 889,900 ****
--- 917,932 ----
  
  static av_cold void X264_init_static(AVCodec *codec)
  {
+ #if X264_BUILD < 153
      if (x264_bit_depth == 8)
          codec->pix_fmts = pix_fmts_8bit;
      else if (x264_bit_depth == 9)
          codec->pix_fmts = pix_fmts_9bit;
      else if (x264_bit_depth == 10)
          codec->pix_fmts = pix_fmts_10bit;
+ #else
+     codec->pix_fmts = pix_fmts_all;
+ #endif
  }
  
  #define OFFSET(x) offsetof(X264Context, x)
